Fix regression that removed function name from stack traces

schmave opened the pull request:
4ec774359d92ceae065151b9c9280211ccc947da unintentionally (?) removed the `name` parameter from stack frames.

PierreQuentel commented:
@schmave 
I removed the 5th item of each element in frames_stack because I don't think it's used anymore (I must confess that I don't remember why it was there) ; the test suite passes without this element so I don't think it's useful to restore it, but maybe I am missing something ?

schmave commented:
I added that 5th item just recently (#856). I used it to include a function name in stack traces. I should have included a test for it in the original PR, but didn't, so I added one here.

PierreQuentel commented:
Sorry, I completely missed this change...

schmave commented:
No problem. Thanks for the merge!
